src/components/Misc/PostButtonModal.tsx

```diff
@@ -0,0 +1,40 @@
+import React, { Dispatch, ReactElement, SetStateAction } from "react";
+import { Dialog, DialogTitle, DialogContent, Button } from "@mui/material/";
+
+const styles = {
+  dialogPosition: {
+    position: "absolute",
+    bottom: "500px",
+  },
+};
+
+type ModalProps = {
+  children: ReactElement<any, any>;
+  openModal: boolean;
+  setOpenModal: Dispatch<SetStateAction<boolean>>;
+};
+
+export default function PostButtonModal(props: ModalProps) {
+  const { children, openModal, setOpenModal } = props;
+  return (
+    <Dialog
+      open={openModal}
+      fullWidth
+      scroll="paper"
+      sx={styles.dialogPosition}
+      PaperProps={{
+        style: {
+          minHeight: "20%",
+          maxHeight: "90%",
+        },
+      }}
+    >
+      <DialogTitle>
+        <Button color="greyButton" onClick={() => setOpenModal(!openModal)}>
```

Comment:
Instead of using a button with an onClick handler, we should use the `onClose` prop on the dialog. They have some examples on MUI's website: https://mui.com/material-ui/react-dialog/#customization 

It not only helps with styling (the padding on the title is huge right now), but also makes it so that when the user clicks away from the dialog it closes too

Along with my `onClose` suggestion above, it would look something like this instead:

```suggestion
        <Dialog onClose={onClose}>
```

Reply:
I have applied this change due to changing the `setOpenModal` prop. It would be a better fit for the reasons you mentioned as well. 

One pushback is I am keeping the onClick handler because onClose on dialog does not affect the close button functionality. I am now calling onClick as `onClick = {onClose}` within the Button attributes.

Reply:
Nice! You're right, we still do need the `onClick` handler. I think we should also use MUI's `IconButton` component rather than their `Button` though